pkg/discovery/dtofactory/group_dto_builder.go

```diff
@@ -42,47 +49,76 @@ func NewGroupDTOBuilder(entityGroupMap map[string]*repository.EntityGroup,
 func (builder *groupDTOBuilder) BuildGroupDTOs() []*proto.GroupDTO {
 	var result []*proto.GroupDTO
 
 	// Groups per parent instance
 	for _, entityGroup := range builder.entityGroupMap {
-		// Pod and containers members belonging to the group
+		if entityGroup.GroupId == "" {
+			glog.Errorf("Invalid group id")
+			continue
+		}
+
+		// Groups for the Pod and containers belonging to the group
 		for etype, memberList := range entityGroup.Members {
-			if entityGroup.GroupId == "" {
-				glog.Errorf("Invalid group id")
-				continue
-			}
+
 			// group id created using the parent type, name and target identifier
 			id := fmt.Sprintf("%s-%s[%s]", entityGroup.GroupId, builder.targetId, etype)
+			displayName := fmt.Sprintf("%ss By %s [%s]", etype, entityGroup.GroupId, builder.targetId)
 
 			var protoType proto.EntityDTO_EntityType
-			if etype == metrics.PodType {
-				protoType = proto.EntityDTO_CONTAINER_POD
-			} else if etype == metrics.ContainerType {
-				protoType = proto.EntityDTO_CONTAINER
-			} else {
-				glog.Errorf("Invalid member entity type")
+			protoType, foundType := ENTITY_TYPE_MAP[etype]
+			if !foundType {
+				glog.Errorf("Invalid member entity type %s", etype)
 				continue
 			}
 
 			// static group
 			groupBuilder := group.StaticGroup(id).
 				OfType(protoType).
-				WithEntities(memberList)
+				WithEntities(memberList).
+				WithDisplayName(displayName)
 
-			// group display name
-			displayName := fmt.Sprintf("%ss By %s [%s]",
-				etype, entityGroup.GroupId, builder.targetId)
-			groupBuilder.WithDisplayName(displayName)
+			// build group
+			groupDTO, err := groupBuilder.Build()
+			if err != nil {
+				glog.Errorf("Error creating group dto  %s::%s", id, err)
+				continue
+			}
+
+			result = append(result, groupDTO)
+
+			glog.V(4).Infof("groupDTO  : %++v", groupDTO)
+		}
+
+		if len(entityGroup.ContainerGroups) <= 1 {
+			continue
+		}
+
+		// Additional sub groups for the different containers running in a pod
+		for containerName, containerList := range entityGroup.ContainerGroups {
+			etype := metrics.ContainerType
+
+			groupId := fmt.Sprintf("%s::%s", entityGroup.GroupId, containerName)
+
+			id := fmt.Sprintf("%s-%s[%s]", groupId, builder.targetId, etype)
+			displayName := fmt.Sprintf("%ss By %s [%s]", etype, groupId, builder.targetId)
+			protoType := proto.EntityDTO_CONTAINER
+
+			// static group
+			groupBuilder := group.StaticGroup(id).
+				OfType(protoType).
+				WithEntities(containerList).
+				WithDisplayName(displayName)
 
-			// group resize policy - currently only for stateful sets
+			// if parent group resize policy is consistent resize, set it for the container sub-group
 			_, exists := CONSISTENT_RESIZE_GROUPS[entityGroup.ParentKind]
 			if exists && entityGroup.ParentName != "" {
-				glog.V(4).Infof("%s: set group to resize consistently\n", entityGroup.GroupId)
+				glog.V(2).Infof("%s: set group to resize consistently\n", groupId)
 				groupBuilder.ResizeConsistently()
```

Comment:
I don't quite understand the code here as well as the comment. 
"if parent group resize policy is consistent resize, set it for the container sub-group"

I think the parent group should always be consistent resize false (if there're more than one container)

Reply:
The consistent resize is set to true only for container sub-groups whose parent type are part of the  CONSISTENT_RESIZE_GROUP map.